File: py/prob_942.py
class Solution:
    def diStringMatch(self, S):
        """
        :type S: str
        :rtype: List[int]
        """
        out_list = []
        low = 0
        high = len(S)
        for char in S:
            if char == 'I':
                out_list.append(low)
                low += 1
            elif char == 'D':
                out_list.append(high)
                high -= 1
        out_list.append(low)
        return out_list

File: cpp/prob_942.cpp
class Solution {
public:
    vector<int> diStringMatch(string S) {
        vector<int> out_list;
        int low = 0;
        int high = S.size();
        for (int char_i=0; char_i<S.size(); char_i++) {
            if (S[char_i] == 'I') {
                out_list.push_back(low);
                low++;
            }
            else if (S[char_i] == 'D') {
                out_list.push_back(high);
                high--;
            }
        }
        out_list.push_back(low);
        return out_list;
    };
};